pandas/core/reshape/util.py

```diff
@@ -51,9 +52,58 @@ def cartesian_product(X):
         # if any factor is empty, the cartesian product is empty
         b = np.zeros_like(cumprodX)
 
-    return [
-        np.tile(
-            np.repeat(np.asarray(com.values_from_object(x)), b[i]), np.product(a[i])
-        )
-        for i, x in enumerate(X)
-    ]
+    return [_tile_compat(np.repeat(x, b[i]), np.product(a[i])) for i, x in enumerate(X)]
+
+
+def _broadcast_tile(arr: np.ndarray, num: int) -> np.ndarray:
+    """
+    Emulate np.tile but using views instead of copies.
```

Comment:
Is this an important optimization? Currently we also use `np.tile` ?

Reply:
For large inputs yes